### PR TITLE
APP-4455 new btn component

### PR DESCRIPTION
**Pull request by symphony-annaLlorens**
### Description 
 - Migration to SDS button

### Demo
Screenshot 2021-09-23 at 14 38 11

Screenshot 2021-09-23 at 13 42 54

### CODE REVIEW

File: src/components/button/Button.tsx [hunk 1]
```diff
@@ -14,9 +14,14 @@ export type ButtonProps = {
   /** If true, substitutes the button content by an animated loader */
   loading?: boolean;
   type?: 'button' | 'reset' | 'submit';
-   /** The variant to use*/
-  variant?: 'primary' | 'secondary' | 'tertiary' | 'destructive';
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+   /** The variant to use*/
+  variant?: 'primary' | 'secondary' | 'tertiary';
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:19]: we should keep "destructive" and have a fallback to work, and be backward compatible

File: src/components/button/Button.tsx [hunk 2]
```diff
@@ -14,9 +14,14 @@ export type ButtonProps = {
...
+   /** Subvariant for each variant */
+  isDesctruvtive?: boolean;
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:21]: typo :)

File: src/components/button/Button.tsx [hunk 3]
```diff
@@ -14,9 +14,14 @@ export type ButtonProps = {
...
+  /** Subvariant for destructive */
+  isActive?: boolean;
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:23]: "isAccent" I think

Comment (AliciaSymphony) [reply src/components/button/Button.tsx:23]: Apart from that, I am wondering is adding two booleans isDestructive and isAccent is the right approach to handle the different variants...
@symphony-jean-michael @symphony-adnane

Comment (AliciaSymphony) [reply src/components/button/Button.tsx:23]: What about variant= "primary-destructive", "tertiary-accent", etc? So we are clear about the variant we are handling

or keep separated like
```
variant: 'primary' | 'secondary' | 'tertiary'
subVariant: "destructive", "accent"
```

File: src/components/button/Button.tsx [hunk 4]
```diff
@@ -14,9 +14,14 @@ export type ButtonProps = {
...
+  size?: 'large' | 'small';
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:24]: also medium? and be the default value

**Review by AliciaSymphony — COMMENTED** [review]
Some comments,
Also, have you noticed that the space between the icon and the text is 8px in medium/large, and 4px in small?
(I cannot see very well in the screenshot, but the spacings look the same)

File: src/components/button/Button.tsx [hunk 5]
```diff
@@ -16,12 +16,8 @@ export type ButtonProps = {
   type?: 'button' | 'reset' | 'submit';
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
    /** The variant to use*/
-  variant?: 'primary' | 'secondary' | 'tertiary';
-   /** Subvariant for each variant */
-  isDesctruvtive?: boolean;
-  /** Subvariant for destructive */
-  isActive?: boolean;
-  size?: 'large' | 'small';
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'destructive' | 'primary--destructive' | 'secondary--destructive'|'tertiary--destructive' |'tertiary--accent';
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:19]: I would go for only one hyphen, it is easier to use :)

File: src/components/button/Button.tsx (hunk 4 above)
Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx]: done

File: src/components/button/Button.tsx [hunk 6]
```diff
@@ -14,9 +14,10 @@ export type ButtonProps = {
   /** If true, substitutes the button content by an animated loader */
   loading?: boolean;
   type?: 'button' | 'reset' | 'submit';
-   /** The variant to use*/
-  variant?: 'primary' | 'secondary' | 'tertiary' | 'destructive';
   onClick?: (event: React.MouseEvent<HTMLElement>) => void;
+   /** The variant to use*/
+  variant?: 'primary' | 'secondary' | 'tertiary' | 'destructive' | 'primary-destructive' | 'secondary-destructive'|'tertiary-destructive' |'tertiary-accent';
```
Comment (AliciaSymphony) [inline src/components/button/Button.tsx:19]: Could you inform that the "destructive" is deprecated please?
With @deprecated annotation, console.warn.... you can check what has been done to EmptyString, or anything if you have suggestions :)

Comment (symphony-annaLlorens) [reply src/components/button/Button.tsx:19]: Screenshot 2021-09-21 at 12 58 24

**Review by symphony-adnane — APPROVED** [review]

**Review by symphony-jean-michael — APPROVED** [review]
LGTM

**Review by symphony-adnane — APPROVED** [review]

**Comment by symphony-annaLlorens** [comment]
ok to test

**Pull request merged**